posts/navarroaxel/16.git-autocorrect.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Git autocorrect needs more marketing
+published: true
+description: How to configure Git to autofix typos in your input commands
+tags: 'git, bash, productivity, tooling'
+cover_image: ./assets/git-autocorrect.png
+id: 1905594
+---
+
+Git comes with a lot of configurations to improve your experience with the command-line. We already talked about how to [color moved blocks of code](https://dev.to/cloudx/how-to-color-the-moved-code-in-git-10ei), but recently I found an awesome configuration called [autocorrect](https://git-scm.com/docs/git-config#Documentation/git-config.txt-helpautoCorrect). 🤩
+
+This feature allows Git to detect typos in your input and display similar valid commands. Git can also suggest the correct one or even run the suggestion automatically. Let's see!
+
+## The default value
+
+By default Git displays suggested commands when we had a typo:
+
+```bash
+$ git pll
+git: 'pll' is not a git command. See 'git --help'.
+
+The most similar command is
+         pull
+```
+
+But we can get more than just a suggestion… 👀
+
+## Check before run the suggestion
+
+If you configure autoCorrect with `prompt` Git will ask for a confirmation to run the suggested command.
+
+```bash
+git config –global help.autocorrect prompt
+WARNING: You called a Git command named 'pll', which does not exist.
+Run 'pull' instead [y/N]?
+```
+
+## Run the suggestion without manual intervention
+
+If you use a numeric value for autoCorrect and Git **finds only 1 valid command** to fix the typo, then the valid one will be executed without confirmation. 🚀
+
+```bash
+$ git config –global help.autoCorrect 30
+$ git pll
+WARNING: You called a Git command named 'pll', which does not exist.
+Continuing in 3 seconds, assuming that you meant 'pull'.
+Already up to date.
+```
+
+💡 You can stop it using `Ctrl + C` or let Git run the assumption.
+
+The numeric value is the timeout in deciseconds (0.1 seconds), but if you want Git to run the fixed command without any wait you should use `immediate`.
+
+🧠 The value `0` disables the auto execution and only displays suggestions.
+
+## Conclusion
+
+If you combine this autocorrect feature with popular aliases you can customize your experience with the command-line to rock on your productivity. 🎸
```

Comment:
```suggestion
If you combine this autocorrect feature with popular aliases you achieve a customized experience with the command-line to rock on your productivity. 🎸
```
This is just a suggestion to avoid the repetition of _you/your_